### PR TITLE
Improve extensibility in Applications

### PR DESCRIPTION
**Pull request by induwaram**
### Purpose
Through this PR extensibility of the applications will be improved . Features such as overriding a tab , customizing a tab , customized label feature ..etc will be added.

### Expected Outcome

- Ability to increase the extensibility of the application
- Overriding a tab
- Customizing a tab

### Related Issues

- N/A

### Related PRs

- N/A

### Checklists

- [ ] e2e cypress tests locally verified.
- [ ] Manual test round performed and verified.
- [ ] UX/UI review done on final implementation.
- [ ] Documentation provided. (Add links if there's any
- [ ]  Unit tests provided. (Add links if there are any)
- [ ] Integration tests provided. (Add links if there are any)

### Security Checks

- [ ] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [ ] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: apps/console/src/features/applications/components/application-danger-zone.tsx [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+     appId?: string;
+     /**
+      * Name of the application.
+      */
+     name: string;
+     /**
+      * Customized content.
+      */
+     content?: string;
+    }
```
Comment (brionmario) [inline apps/console/src/features/applications/components/application-danger-zone.tsx:46]: Fix indentation.

File: apps/console/src/features/applications/components/application-danger-zone.tsx [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+    
+/**
+ * Application Danger Zone component
+ *
```
Comment (brionmario) [inline apps/console/src/features/applications/components/application-danger-zone.tsx:50]:
```suggestion
 * @param { ApplicationDangerZonePropsInterface } props - Props injected to the component.
```

File: apps/console/src/features/applications/components/application-danger-zone.tsx [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+ * @return {JSX.Element}
```
Comment (brionmario) [inline apps/console/src/features/applications/components/application-danger-zone.tsx:51]:
```suggestion
 * @return {React.ReactElement}
```

File: apps/console/src/features/applications/components/application-danger-zone.tsx [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+        [ "data-componentid" ]: testId
+    } = props;
+     
+    const allowedScopes: string = useSelector((state: AppState) => state?.auth?.allowedScopes);
+    const UIConfig: UIConfigInterface = useSelector((state: AppState) => state?.config?.ui);
+ 
+    const [ showDeleteConfirmationModal, setShowDeleteConfirmationModal ] = useState<boolean>(false);
+    const [ isDeletionInProgress, setIsDeletionInProgress ] = useState<boolean>(false);
+ 
+
```
Comment (brionmario) [inline apps/console/src/features/applications/components/application-danger-zone.tsx:70]: Remove un-neccessary lines.

File: apps/console/src/features/applications/components/application-danger-zone.tsx [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+    } = props;
+     
+    const allowedScopes: string = useSelector((state: AppState) => state?.auth?.allowedScopes);
+    const UIConfig: UIConfigInterface = useSelector((state: AppState) => state?.config?.ui);
+ 
+    const [ showDeleteConfirmationModal, setShowDeleteConfirmationModal ] = useState<boolean>(false);
+    const [ isDeletionInProgress, setIsDeletionInProgress ] = useState<boolean>(false);
+ 
+ 
+    const { t } = useTranslation();
```
Comment (brionmario) [inline apps/console/src/features/applications/components/application-danger-zone.tsx:71]: Move hooks to the top just after props.

```tsx
    const {
        featureConfig,
        appId,
        name,
        content,
        [ "data-componentid" ]: testId
    } = props;
    
    const dispatch = useDispatch();
    
    const { t } = useTranslation();

```

Proper hierachy.

```
props
custom hooks
redux selctors
usestate
useEffect
functions
return
```

File: apps/console/src/features/applications/components/edit-application.tsx [hunk 6]
```diff
@@ -736,72 +734,113 @@ export const EditApplication: FunctionComponent<EditApplicationPropsInterface> =
...
-                });
+                applicationConfig.editApplication.
+                    isTabEnabledForApp(inboundProtocolConfig?.oidc?.clientId, ApplicationTabTypes.USER_ATTRIBUTES) &&
+                  panes.push({
+                      componentId: "user-attributes",
+                      menuItem: 
+                         <Menu.Item data-tourid="attributes">
+                             { t("console:develop.features.applications.edit.sections.attributes.tabName") }
+                         </Menu.Item>,
+                      render:() =>
```
Comment (brionmario) [inline apps/console/src/features/applications/components/edit-application.tsx:779]:
```suggestion
                      render: () => 
```

File: apps/console/src/features/applications/components/application-danger-zone.tsx (hunk 1 above)
Comment (induwaram) [reply apps/console/src/features/applications/components/application-danger-zone.tsx:46]: Fixed with https://github.com/wso2/identity-apps/pull/3042/commits/39901f546e17bf0fae6e1dfba50f7c76d0369360

File: apps/console/src/features/applications/components/application-danger-zone.tsx (hunk 2 above)
Comment (induwaram) [reply apps/console/src/features/applications/components/application-danger-zone.tsx]: Fixed with https://github.com/wso2/identity-apps/pull/3042/commits/39901f546e17bf0fae6e1dfba50f7c76d0369360

File: apps/console/src/features/applications/components/application-danger-zone.tsx (hunk 4 above)
Comment (induwaram) [reply apps/console/src/features/applications/components/application-danger-zone.tsx]: Fixed with https://github.com/wso2/identity-apps/pull/3042/commits/39901f546e17bf0fae6e1dfba50f7c76d0369360

File: apps/console/src/features/applications/components/application-danger-zone.tsx (hunk 5 above)
Comment (induwaram) [reply apps/console/src/features/applications/components/application-danger-zone.tsx]: Fixed with https://github.com/wso2/identity-apps/pull/3042/commits/39901f546e17bf0fae6e1dfba50f7c76d0369360

File: apps/console/src/features/applications/components/edit-application.tsx (hunk 6 above)
Comment (induwaram) [reply apps/console/src/features/applications/components/edit-application.tsx]: Fixed with https://github.com/wso2/identity-apps/pull/3042/commits/39901f546e17bf0fae6e1dfba50f7c76d0369360

File: apps/console/src/features/applications/components/application-danger-zone.tsx [hunk 7]
```diff
@@ -0,0 +1,210 @@
...
+import { addAlert } from "@wso2is/core/store";
+import { ConfirmationModal, DangerZone, DangerZoneGroup } from "@wso2is/react-components";
+import React, { FunctionComponent, ReactElement, useState } from "react";
+import { useTranslation } from "react-i18next";
+import { useDispatch, useSelector } from "react-redux";
+import { AppConstants, AppState, FeatureConfigInterface, UIConfigInterface, history } from "../../core";
+import { deleteApplication } from "../api";
+   
+/**
+ * Prop types of the `LoginPlaygroundWizard` component.
```
Comment (brionmario) [inline apps/console/src/features/applications/components/application-danger-zone.tsx:30]:
```suggestion
 * Prop types of the  ApplicationDangerZone component.
```

**Review by brionmario — APPROVED** [review]

File: apps/console/src/features/applications/components/application-danger-zone.tsx [hunk 8]
```diff
@@ -0,0 +1,210 @@
+/**
+ * Copyright (c) 2022, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (brionmario) [inline apps/console/src/features/applications/components/application-danger-zone.tsx:2]:
```suggestion
 * Copyright (c) 2022, WSO2 Inc. (http://www.wso2.com) All Rights Reserved.
```

**Pull request merged**